web/modules/custom/genre_validator/src/Plugin/Validation/Constraint/genreRulesValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+
+namespace Drupal\genre_validator\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the genre_rules Constraint
+ *
+ * @package Drupal\genre_validator\Plugin\Validation\Constraint
+ */
+class genreRulesValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+
+  public function validate($genres, Constraint $constraint) {
+
+    foreach ($genres as $genre) {
+      // Check if the string has any numbers
+      if (preg_match('/[123456789]/', $genre)) {
+        $this->context->addViolation($constraint->noNumbers, ['%value' => $genre->value]);
+      }
+      // Check if the string has any special characters
+      if (preg_match('/[\'^£$%&*()}{@#~?<>,|=_+¬-]/', $genre)) {
```

Comment:
Not all characters are here... try something like `\W`

Reply:
I just remembered that dashes shouldn't be excluded. So would [\W]+[^-] be okay here?